### PR TITLE
[HW-01] Set up express/webpack/react app

### PR DESCRIPTION
**Pull request by tunya**
The one thing bothering me that I couldn't fix is that when I run webpack it creates style.js alongside with style.css and includes it to final html file.
Do you have any thoughts on how I can fix it?

### CODE REVIEW

File: app/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const accomplished = [
+    'Set up the dev environment for React Mentoring Program',
+    'Implemented simple React + Webpack app',
+    'Added dependencies needed for furhter work'
+];
+
+var element1 = <main>
```
Comment (Aksana-Tsishchanka) [inline app/src/index.jsx]: According to ES6 you have to avoid "var" using.

File: app/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <h1>Homework 1</h1>
+    <ul>
+        {accomplished.map((elem) => <li>{elem}</li>)}
```
Comment (Aksana-Tsishchanka) [inline app/src/index.jsx]: unnecessary parentheses in callback arguments: 
elem => <li>{elem}</li>

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const path = require('path');
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js:1]: add "clean-webpack-plugin", to remove build folder before new bundle building

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+const webpack = require('webpack');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+
+const BUILD_DIR  = path.join(__dirname, 'app/build');
+const CLIENT_DIR = path.join(__dirname, 'app/src');
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js]: rename CLIENT_DIR to SOURCE_DIR

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+module.exports = {
+    context: CLIENT_DIR,
+
+    entry: {
+        index: [
+            './index',
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js]: 1) let's not complicate webpack config and just keep only entry file './index.jsx'. Do you need "resolve" section now?
2) webpack-dev-server is used only for development, as you know, on project are 2 mode: production and development. So I suggest in npm scripts have smth like "npm run dev" and npm run build".
In development mode you have to run webpack-dev-server with options if any.
Please follow this guide https://webpack.js.org/guides/development/#using-webpack-dev-server

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            'react-hot-loader/patch',
+            'webpack-dev-server/client?http://localhost:3000'
+        ],
+        style: './style.sass'
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js]: As we already clarified, this style entry has to be removed and imported in js/jsx files.

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            test: /\.sass$/,
+            use: ExtractTextPlugin.extract({
+                fallback: 'style-loader',
+                use: ['css-loader', 'sass-loader']
+            })
+        }]
+    },
+
+    plugins: [
+        new webpack.HotModuleReplacementPlugin(),
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js]: It does not work, changes are not applied automatically. I suggest use webpack-dev-server and it's enough for development in this course.

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        new webpack.NamedModulesPlugin(),
+        new HtmlWebpackPlugin({
+            title: 'Homework 1',
+            hash: true,
+            template: './index.html'
+        }),
+        new ExtractTextPlugin({
```
Comment (Aksana-Tsishchanka) [inline webpack.config.js:51]: It's good approach to keep all styles in one, when app is on production.
But for development is not necessary, you will develop components and usually each component has to have own style. In js/jsx file you have at least 2 options to work with styles:
1) import './style.sass'. As a result, you have **Embedded CSS**, it's fine in development and hot reloading with webpack-dev-server has to work perfectly.
2) CSS Modules, which allows to set styles locally! Read https://github.com/css-modules/css-modules

File: package.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  "version": "0.0.1",
+  "author": "Tatiana Tun",
+  "license": "ISC",
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/tunya/react-mp17.git"
+  },
+  "scripts": {
+    "start": "node app/server/index.js",
+    "dev": "webpack && npm start"
```
Comment (Aksana-Tsishchanka) [inline package.json]: you need:
1) run development mode (webpack-dev-server is running, there are no build folder) -> npm run dev
2) run let's say "production mode" (build folder has to be generated with all necessary files) -> npm run build

File: app/src/index.jsx (hunk 1 above)
Comment (tunya) [reply app/src/index.jsx]: Fixed

File: app/src/index.jsx (hunk 2 above)
Comment (tunya) [reply app/src/index.jsx]: Fixed

File: webpack.config.js (hunk 3 above)
Comment (tunya) [reply webpack.config.js:1]: Done

File: webpack.config.js (hunk 4 above)
Comment (tunya) [reply webpack.config.js]: Fixed

File: webpack.config.js (hunk 5 above)
Comment (tunya) [reply webpack.config.js]: Done

File: webpack.config.js (hunk 7 above)
Comment (tunya) [reply webpack.config.js]: Removed it and used dev-server instead

File: package.json (hunk 9 above)
Comment (tunya) [reply package.json]: Done

**Pull request merged**